src/child/components/Favourite.js

```diff
@@ -86,18 +84,39 @@ class Favourite extends Component {
         const percentage = !isNaN(+stockData.percentage) ? Math.abs((+stockData.percentage)).toFixed(2) : '';
         const name = stockData.name ? truncate(stockData.name) : '';
 
+        const flipModal = starTopPos => {
+            const modalBottom = starTopPos + modalOffset + modalHeight;
+            return modalBottom >= window.innerHeight;
+        };
+
         return (
             <div
               id={`stock_${stockCode}`}
-              draggable="true"
+              draggable={!isUnfavouriting}
               className="favouriteWrapper"
               onClick={() => bindings.onClick(stockCode, name)}
               onDoubleClick={() => bindings.onDoubleClick(stockCode, name)}
             >
                 <div className="drop-target">
                     <div className={`darkens favourite tearable ${cls}`} draggable="false">
                         <div className="top">
-                            <div className="button-icon star active" onClick={this.onIconClick}>&nbsp;</div>
+                            <div ref="star" className="button-icon star active" onClick={bindings.onIconClick(stockCode)}>&nbsp;</div>
+                            {isUnfavouriting ?
+                                <Confirmation
```

Comment:
Is it possible to tidy up this section by passing functions down through bindings just?
I like that the `Confirmation` component is kept dumb, but it makes the `Favourite` looks pretty messy in my opinion